format/spec.md

```diff
@@ -158,7 +158,7 @@ For the representations of these types in Avro, ORC, and Parquet file formats, s
 
 #### Nested Types
 
-A **`struct`** is a tuple of typed values. Each field in the tuple is named and has an integer id that is unique in the table schema. Each field can be either optional or required, meaning that values can (or cannot) be null. Fields may be any type. Fields may have an optional comment or doc string.
+A **`struct`** is a tuple of typed values. Each field in the tuple is named and has an integer id that is unique in the table schema. Each field can be either optional or required, meaning that values can (or cannot) be null. Fields may be any type. Fields may have an optional comment or doc string. Fields can also have a default value (see below).
```

Comment:
Do we need to specify that these default values are only retroactive? Ie we still do not allow skipping a required column just because the said column has a default? I haven't read the SQL spec on this or other system's usage, but I assume we want to align on that.

Reply:
@RussellSpitzer are you saying that all columns are required at write time? It sounds like that's the case. Just because there's a default value doesn't mean you can skip writing a column to a Parquet file. They must be present.

I think that's a good thing to call out!

Reply:
This sounds like the `INSERT INTO` use case in [this comment](https://github.com/apache/iceberg/pull/2496#issuecomment-1048352760). The support should come from the engine first. Regardless, yes, we cannot support that today.